### PR TITLE
fix(mempool): pending withdrawals not in mem block aren't restored

### PR DESCRIPTION
**Pull request by zeroqn**
- restore pending withdrawals from db

### CODE REVIEW

**Review by jjyr — DISMISSED** [review]

**Comment by Flouse** [comment]
What is the plan of `TODO: Check replace-by-fee` in `fn push_withdrawal_request`?
@jjyr

File: crates/tests/src/tests/restore_mem_pool_pending_withdrawal.rs [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+                    .owner_lock(owner_lock)
+                    .build()
+            })
+            .collect()
+    };
+    let pending_withdrawals = withdrawals.split_off(withdrawals.len() - 5);
+    let mem_block_withdrawals = withdrawals;
+
+    // Insert error nonce withdrawal and expect them to be removed during pending restore
+    let invalid_withdrawals_count = pending_withdrawals.len() - 5;
```
Comment (Flouse) [inline crates/tests/src/tests/restore_mem_pool_pending_withdrawal.rs:100]: `invalid_withdrawals_count` here would be zero?

**Review by Flouse — CHANGES_REQUESTED** [review]

File: crates/tests/src/tests/restore_mem_pool_pending_withdrawal.rs (hunk 1 above)
Comment (zeroqn) [reply crates/tests/src/tests/restore_mem_pool_pending_withdrawal.rs:100]: This count is used to create invalid withdrawals for test, should not be zero.

Comment (zeroqn) [reply crates/tests/src/tests/restore_mem_pool_pending_withdrawal.rs:100]: fixed

**Review by Flouse — APPROVED** [review]

**Review by blckngm — APPROVED** [review]

**Comment by jjyr** [comment]
> What is the plan of `TODO: Check replace-by-fee` in `fn push_withdrawal_request`? @jjyr

We should implement a similar mechanism like TxFeeQueue for the withdrawal

**Pull request merged**